examples/src/NestedProvidersExample.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+import {withDataProviders} from 'data-provider'
+import {connect} from 'react-redux'
+import {compose} from 'redux'
+import {postProvider} from './dataProviders'
+import {togglePost} from './actions'
+
+const Post = ({title, body}) => (
+  <div className="message">
+    <h3>{title}</h3>
+    <p>{body}</p>
+  </div>
+)
+
+// PostContainer uses a data-provider with needed === true
+// - order of connect / withDataProviders is important here, since when data is fetched
+// and dispatched by parent DP, this connect component needs to react to state change - it wouldn't be able to,
+// if it was "under" this DP, because it didn't exist yet
```

Comment:
Hmm, I don't think that's the reason it wouldn't work. If you passed the required props (`title, body`) to the `PostContainer` from `TogglablePostContainer`, it should work - you wouldn't need the `connect` here at all. However, if the order was just reversed now, the `TogglablePostContainer` (and hence `PostContainer`) is never redrawn because the props of  `TogglablePostContainer` don't change.

I'd say that the `withDataProviders` needs to be under the `connect`, so that it's notified of the change.

Reply:
I have removed the comment and reversed the order of `connect` / `withDataProviders` in the latest version of `keep_alive` branch (but it isn't really important anymore), since the refresh logic is completely rewritten there.